packages/auth-solid-start/src/server/index.ts

```diff
@@ -0,0 +1,987 @@
+import {
+  Auth,
+  BuiltinOAuthProviderNames,
+  builtinOAuthProviderNames,
+  ConfigurationError,
+  EdgeDBAuthError,
+  InvalidDataError,
+  OAuthProviderFailureError,
+  PKCEError,
+  SignupResponse,
+  type TokenData,
+} from "@edgedb/auth-core";
+import { type APIEvent } from "@solidjs/start/server";
+import type { Client } from "edgedb";
+import { deleteCookie, getCookie, HTTPEvent, setCookie } from "vinxi/http";
+import {
+  BuiltinProviderNames,
+  SolidAuthHelpers,
+  SolidAuthOptions,
+} from "../shared";
+
+export * from "@edgedb/auth-core/errors";
+export { SolidAuthHelpers };
+export type { BuiltinProviderNames, SolidAuthOptions, TokenData };
+
+type ParamsOrError<
+  Result extends object,
+  ErrorDetails extends object = object,
+> =
+  | ({ error: null } & { [Key in keyof ErrorDetails]?: undefined } & Result)
+  | ({ error: Error } & ErrorDetails & { [Key in keyof Result]?: undefined });
+
+export class SolidAuthSession {
+  public readonly client: Client;
+
+  /** @internal */
+  constructor(
+    client: Client,
+    public readonly authToken: string | null,
+  ) {
+    this.client = this.authToken
+      ? client.withGlobals({ "ext::auth::client_token": this.authToken })
+      : client;
+  }
+
+  private _isSignedIn: Promise<boolean> | null = null;
+  async isSignedIn(): Promise<boolean> {
+    if (!this.authToken) return false;
+    return (
+      this._isSignedIn ??
+      (this._isSignedIn = this.client
+        .queryRequiredSingle<boolean>(
+          `select exists global ext::auth::ClientTokenIdentity`,
+        )
+        .catch(() => false))
+    );
+  }
+}
+
+export interface CreateAuthRouteHandlers {
+  onOAuthCallback(
+    params: ParamsOrError<{
+      tokenData: TokenData;
+      provider: BuiltinOAuthProviderNames;
+      isSignUp: boolean;
+    }>,
+  ): Promise<never | Response>;
```

Comment:
Oh actually, you can type these return values as `Promise<Response>` since returning `Promise<never>` will always match `Promise<Response>` due to ["`never` is assignable to everything."](https://www.typescriptlang.org/docs/handbook/type-compatibility.html#any-unknown-object-void-undefined-null-and-never-assignability) That means that if your function _does_ return `never`, it'll still type check here.